tsup.config.ts

```diff
@@ -1,59 +1,80 @@
 import { defineConfig, Options } from 'tsup'
+import * as glob from 'glob'
 import {
   getWorkerChecksum,
   copyWorkerPlugin,
 } from './config/plugins/esbuild/copyWorkerPlugin'
+import { resolveCoreImportsPlugin } from './config/plugins/esbuild/resolveCoreImportsPlugin'
 
-// Prevent from bunlding the "@mswjs/*" packages
-// so that the users get the latest versions without
-// having to bump them in "msw'."
+// Externalize the in-house dependencies so that the user
+// would get the latest published version automatically.
 const ecosystemDependencies = /^@mswjs\/(.+)$/
 
+// Externalize the core functionality (reused across environments)
+// so that it can be shared between the environments.
+const mswCore = /\/core\/(.+)$/
+
 const SERVICE_WORKER_CHECKSUM = getWorkerChecksum()
 
-const nodeConfig: Options = {
-  name: 'node',
-  platform: 'node',
-  entry: ['./src/index.ts', './src/node/index.ts'],
-  inject: ['./config/polyfills-node.ts'],
+const coreConfig: Options = {
+  name: 'core',
+  platform: 'neutral',
+  entry: glob.sync('./src/core/**/*.ts', {
+    ignore: '**/*.test.ts',
+  }),
```

Comment:
clever

Reply:
I was surprised to learn esbuild doesn't support negative patterns and doesn't have an `ignore` option.